HandleNHCreateOrUpdateInstallationCallActivity/__tests__/handler.test.ts

```diff
@@ -80,16 +71,21 @@ describe("HandleNHCreateOrUpdateInstallationCallActivity", () => {
       notificationHubConfig: aNHConfig
     });
 
-    expect.assertions(2);
+    expect.assertions(3);
 
-    await handler(contextMock as any, input);
+    const res = await handler(contextMock as any, input);
+    expect(res.kind).toEqual("SUCCESS");
 
-    expect(buildNHServiceMock).toHaveBeenCalledTimes(1);
-    expect(buildNHServiceMock).toBeCalledWith(aNHConfig);
+    expect(mockBuildNHService).toHaveBeenCalledTimes(1);
+    expect(mockBuildNHService).toBeCalledWith(aNHConfig);
```

Comment:
Rule of thumb: expectations on internal behaviours have little value in terms of confidence the tested procedure is robust.

We should focus more on the `handler` result. We can play with `mockBuildNHService`, mocking its output to simulate multiple scenarios and then check how `handler` responds to them.

Reply:
>Rule of thumb: expectations on internal behaviours have little value in terms of confidence that tested procedure is robust.

Ok, I get it..